sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -231,7 +233,7 @@ class Iso8601TimestampFormatter(
 
   override def format(instant: Instant): String = {
     try {
-      formatter.withZone(zoneId).format(instant)
+      zonedFormatter.format(instant)
```

Comment:
So, this is a performance improvement? 
- If this is perf improvement, we don't need a test case.
- However, the title could be misleading because it sounds like a bug. :) Could you revise it like improvement?

Reply:
Thank you @dongjoon-hyun 

I have revised the PR title to avoid using bug-like expressions